### ISSUE TITLE
MultiPolygon returned in Union should have first element as parent polygon. 

### ISSUE THREAD

**Issue by Sagarpreet** [w8r/martinez#108]
MultiPolygon returned in Union should have first element as parent polygon. 
So, 
First polygon = main parent polygon . 
After that = all the holes . 

Currently, parent polygon and holes comes in random order .

**Comment by Bronson-Brown-deVost** [comment]
I think I might have a similar problem (but downstream in the Rust package).  It seems that each ring of a polygon (both the exterior and the interior rings) gets treated as a separate polygon in the multipolygon output of the boolean operation.  For example:
```rust
let geomPoly = polygon!(
        exterior: [
            (x: 0.0, y: 0.0),
            (x: 0.0, y: 10.0),
            (x: 10.0, y: 10.0),
            (x: 10.0, y: 0.0),
            (x: 0.0, y: 0.0)],
        interiors: [
            [
            (x: 3.0, y: 3.0),
            (x: 3.0, y: 6.0),
            (x: 6.0, y: 6.0),
            (x: 6.0, y: 3.0),
            (x: 3.0, y: 3.0)]
        ]
    );

    let poly = polygon![
        (x: 1.0, y: 1.0),
        (x: 4.0, y: 1.0),
        (x: 4.0, y: 4.0),
        (x: 1.0, y: 4.0),
        (x: 1.0, y: 1.0),
    ];

    let union = geomPoly.union(&poly);
``` 
The union is (in WKT) `MULTIPOLYGON(((0 0,10 0,10 10,0 10,0 0)),((3 4,4 4,4 3,6 3,6 6,3 6,3 4)))` which should be `MULTIPOLYGON(((0 0,10 0,10 10,0 10,0 0),(3 4,4 4,4 3,6 3,6 6,3 6,3 4)))`.  That is, the output is correct except for the nesting; the rings are treated as separate polygons, rather than the second ring being an inner ring for the parent.  I can perhaps add an example in JS if that would be helpful (or maybe this upstream JS library doesn't have the problem).
Thanks very much for the lovely package.

**Comment by bluenote10** [comment]
@Bronson-Brown-deVost For the Rust version this has already be fixed in https://github.com/21re/rust-geo-booleanop/pull/10:

![image](https://user-images.githubusercontent.com/3620703/77572680-af816580-6ecf-11ea-9a2a-29d0a56e6f94.png)

We still have do a Cargo release, but until then you can use the version directly from master.

For JS the same is implemented in #113 and ready to merge.

**Comment by Bronson-Brown-deVost** [comment]
Many thanks, it works great!
